### PR TITLE
Add thread-safety note for NetCDF4

### PR DESCRIPTION
**Pull request by danlooo**
see #450

### CODE REVIEW

**Comment by felixcremer** [comment]
Could we make this documentation a bit more explicit and show on an example how to use these locks for threadsafe opening a NetCDF?

File: docs/src/UserGuide/read.md [hunk 1]
```diff
@@ -43,6 +43,9 @@ Individual arrays can be accessed using subsetting:
 ds.tos
 ````
 
+Please note that netCDF4 uses HDF5 which is not thread-safe in Julia.
+Use manual [locks](https://docs.julialang.org/en/v1/manual/multi-threading/#man-using-locks) in your own code to avoid any data-race.
```
Comment (lazarusA) [inline docs/src/UserGuide/read.md:47]: maybe could you put here, the how? the link is fine but having the solution here would be nice 😄 .

Comment (danlooo) [reply docs/src/UserGuide/read.md]: I added an example in [c9f55a2](https://github.com/JuliaDataCubes/YAXArrays.jl/pull/451/commits/c9f55a2045123b6514740f03879aa9c3ba6e35fc). Simple enough for demonstration but a little bit silly. Is there any nice real-world example of this?

Comment (lazarusA) [reply docs/src/UserGuide/read.md]: mmm... it looks like https://github.com/JuliaIO/HDF5.jl/pull/1021  fixed this, weird that still fails here. However, is good to know how to address this if seeing the wild. Thanks!

**Review by lazarusA — APPROVED** [review]

**Pull request merged**